vmware-tanzu/tanzu-framework
# Packagify autoscaler

### What this PR does / why we need it

PR to convert autoscaler into a package and install it onto the management cluster when a workload cluster is being created

### Which issue(s) this PR fixes


Fixes #https://github.com/vmware-tanzu/tanzu-framework/issues/2320

### Describe testing done for PR
Manual Testing
1. Created and published the autoscaler package to my personal registry
2. Created a management-cluster on AWS
3. Set ENABLE_AUTOSCALER to true and created a workload cluster
4. Workload cluster creation was successful
5. Verified that the autoscaler deployment came up on the workload cluster
```
asudarshan-a01 ~/.config/tanzu/tkg/providers/ytt/03_customizations/autoscaler $ kubectl get pkgi -A
NAMESPACE    NAME                            PACKAGE NAME                               PACKAGE VERSION                            DESCRIPTION                                                            AGE
default      aws-cc-pi                       tkg-clusterclass-aws.tanzu.vmware.com      0.18.0-dev                                 Reconcile succeeded                                                    3d11h
default      test-wc-2-kapp-controller       kapp-controller.tanzu.vmware.com           0.34.0+vmware.1-tkg.1-zshippable           Delete failed: Error (see .status.usefulErrorMessage for details)      3h14m
default      test-wc-3-cluster-autoscaler    tkg-autoscaler.tanzu.vmware.com            0.21.0-dev                                 Reconcile succeeded                                                    2m15s
...


kubectl get deployments -A
NAMESPACE                           NAME                                            READY   UP-TO-DATE   AVAILABLE   AGE
capa-system                         capa-controller-manager                         1/1     1            1           3d11h
capi-kubeadm-bootstrap-system       capi-kubeadm-bootstrap-controller-manager       1/1     1            1           3d11h
capi-kubeadm-control-plane-system   capi-kubeadm-control-plane-controller-manager   1/1     1            1           3d11h
capi-system                         capi-controller-manager                         1/1     1            1           3d11h
cert-manager                        cert-manager                                    1/1     1            1           3d11h
cert-manager                        cert-manager-cainjector                         1/1     1            1           3d11h
cert-manager                        cert-manager-webhook                            1/1     1            1           3d11h
default                             test-wc-3-cluster-autoscaler                    1/1     1            1           2m28s
``` 



### Release note

```release-note
* Creates scaffolding for the autoscaler package
* Installs autoscaler on the management-cluster while a workload cluster is being created if the user enables it
```

### PR Checklist



      


### Additional information

#### Special notes for your reviewer
